api/src/main/java/org/openmrs/module/aijar/dataintegrity/BasePatientRuleDefinition.java

```diff
@@ -41,6 +49,39 @@ public String getExposedInfantNumber(Patient patient) {
 		PatientIdentifierType pit = patientService.getPatientIdentifierTypeByUuid(PatientIdentifierTypes.EXPOSED_INFANT_NUMBER.uuid());
 		return patient.getPatientIdentifier(pit) == null ? "" : patient.getPatientIdentifier(pit).toString();
 	}
+
+	/**
+	 * TB number for the patient
+	 * @param patient
+	 * @return
+	 */
+	public String getTbNumber(Patient patient) { //To be done
```

Comment:
@moshonk  You do not need the most recent number but rather the TB number associated with the summary encounter.. If a patient has 3 different TB treatments then you need to show the correct identifiers for encounter being shown .. You should tie to an encounter since there is only one identifier per encounter